packages/react-portal/src/usePortalMountNode.ts

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { useIsomorphicLayoutEffect, isSSR } from '@fluentui/react-utilities';
+import { useThemeClassName, useFluent } from '@fluentui/react-shared-contexts';
+
+export type UsePortalElementOptions = {
+  /**
+   * The document where the portal element will be created
+   *
+   * @defaultValue document from Fluent context
+   */
+  targetDocument?: Document | undefined;
+  /**
+   * HTML dir attribute
+   *
+   * @defaultVaue dir from Fluent context
+   */
+  dir?: React.HTMLAttributes<HTMLElement>['dir'];
+  /**
+   * CSS class to apply
+   */
+  className?: string;
+
+  /**
+   * Since hooks cannot be called conditionally use this flag to disable creating the node
+   */
+  disable?: boolean;
+};
+
+/**
+ * Creates a new element on a document.body to mount portals
+ */
+export const usePortalMountNode = (options: UsePortalElementOptions) => {
+  const themeClassName = useThemeClassName();
+  const { document: fluentDocument, dir: fluentDir } = useFluent();
```

Comment:
```suggestion
  const { targetDocument: fluentDocument, dir: fluentDir } = useFluent();
```

#17827 will introduce renaming.

Reply:
Will change this in that PR